web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -95,6 +95,28 @@ export class DataStoreService {
     return await Promise.all(querySnapshot.docs.map(doc => doc.ref.delete()));
   }
 
+  async deleteFeature(projectId: string, featureId: string) {
+    await this.deleteAllObservationsInFeature(projectId, featureId);
+    return await this.db
```

Comment:
Since .delete() itself returns a Promise, I think this `await` is redundant (it will unwrap the result from the Promise, which will get rewrapped by being returned from an `async` function.